cmd/collector/app/span_processor_test.go

```diff
@@ -346,3 +347,160 @@ func TestSpanProcessorWithCollectorTags(t *testing.T) {
 		assert.True(t, foundTag)
 	}
 }
+
+func TestSpanProcessorCountSpan(t *testing.T) {
+	mb := metricstest.NewFactory(time.Hour)
+	m := mb.Namespace(metrics.NSOptions{})
+
+	w := &fakeSpanWriter{}
+	p := NewSpanProcessor(w, Options.HostMetrics(m), Options.DynQueueSizeWarmup(1000)).(*spanProcessor)
+	p.background(10*time.Millisecond, p.updateGauges)
+	defer p.Stop()
+
+	p.processSpan(&model.Span{})
+	assert.NotEqual(t, uint64(0), p.bytesProcessed)
+
+	for i := 0; i < 15; i++ {
+		_, g := mb.Snapshot()
+		if b := g["spans.bytes"]; b > 0 {
+			assert.Equal(t, p.bytesProcessed.Load(), uint64(g["spans.bytes"]))
+			return
+		}
+		time.Sleep(time.Millisecond)
+	}
+
+	assert.Fail(t, "gauge hasn't been updated within a reasonable amount of time")
+}
+
+func TestUpdateQueueSizeScaleUp(t *testing.T) {
```

Comment:
Could some of the following tests be DRYer through table-driven tests?

Reply:
Good idea!